src/main/java/jcifs/smb1/smb1/SmbTransport.java

```diff
@@ -0,0 +1,820 @@
+
+                req.maxBufferSize = snd_buf_size;
+                resp.reset();
+
+                try {
+                    BufferCache.getBuffers( req, resp );
+
+                    /* 
+                     * First request w/ interim response
+                     */ 
+
+                    req.nextElement();
+                    if (req.hasMoreElements()) {
+                        SmbComBlankResponse interim = new SmbComBlankResponse();
+                        super.sendrecv( req, interim, RESPONSE_TIMEOUT );
+                        if (interim.errorCode != 0) {
+                            checkStatus( req, interim );
+                        }
+                        req.nextElement();
+                    } else {
+                        makeKey( req );
+                    }
+
+                    synchronized (this) {
+                        response.received = false;
+                        resp.isReceived = false;
+                        try {
+                            response_map.put( req, resp );
+
+                            /* 
+                             * Send multiple fragments
+                             */
+
+                            do {
+                                doSend0( req );
+                            } while( req.hasMoreElements() && req.nextElement() != null );
+
+                            /* 
+                             * Receive multiple fragments
+                             */
+
+                            long timeout = RESPONSE_TIMEOUT;
+                            resp.expiration = System.currentTimeMillis() + timeout;
+                            while( resp.hasMoreElements() ) {
+                                wait( timeout );
+                                timeout = resp.expiration - System.currentTimeMillis();
+                                if (timeout <= 0) {
+                                    throw new TransportException( this +
+                                            " timedout waiting for response to " +
+                                            req );
+                                }
+                            }
+                            if (response.errorCode != 0) {
+                                checkStatus( req, resp );
+                            }
+                        } catch( InterruptedException ie ) {
+                            throw new TransportException( ie );
+                        } finally {
+                            response_map.remove( req );
+                        }
+                    }
+                } finally {
+                    BufferCache.releaseBuffer( req.txn_buf );
+                    BufferCache.releaseBuffer( resp.txn_buf );
+                }
+
+            } else {
+                response.command = request.command;
+                super.sendrecv( request, response, RESPONSE_TIMEOUT );
+            }
+        } catch( SmbException se ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)